src/os/bluestore/BlueStore.cc

```diff
@@ -6583,6 +6810,56 @@ int BlueStore::_check_or_set_bdev_label(
   return 0;
 }
 
+int BlueStore::_set_main_bdev_label()
+{
+  uint64_t size = bdev->get_size();
+  bdev_label_valid_locations.clear();
+  bluestore_bdev_label_t& label = bdev_label;
+  label.osd_uuid = fsid;
+  label.size = size;
+  label.btime = ceph_clock_now();
+  label.description = "main";
+  if (cct->_conf.get_val<bool>("bluestore_bdev_label_multi")) {
+    label.meta["multi"] = "yes";
+    label.meta["epoch"] = "1";
+    bdev_label_multi = true;
+    bdev_label_epoch = 1;
+    for (uint64_t position : bdev_label_positions) {
+      if (position + BDEV_LABEL_BLOCK_SIZE <= size) {
+        bdev_label_valid_locations.push_back(position);
+      }
+    }
+  } else {
+    bdev_label_valid_locations.push_back(BDEV_FIRST_LABEL_POSITION);
+  }
+  int r = _write_bdev_label(cct, path + "/block", label, bdev_label_valid_locations);
+  if (r < 0)
+    return r;
+  return 0;
+}
+
+int BlueStore::_check_main_bdev_label()
+{
+  string block_path = path + "/block";
+  int r = _read_main_bdev_label(&bdev_label, &bdev_label_valid_locations,
+    &bdev_label_multi, &bdev_label_epoch);
+  if (r < 0)
+    return r;
+  if (cct->_conf->bluestore_debug_permit_any_bdev_label) {
```

Comment:
Does this still make any sense? Previously we could permit using a label having arbitrary uuid by tricking with this parameter. Is this still applicable?

Reply:
Modified `_read_main_bdev_label` to conform to logic of `bluestore_debug_permit_any_bdev_label`.